wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/flywheelbangbangcontroller/Robot.java

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj.examples.flywheelbangbangcontroller;
+
+import edu.wpi.first.math.controller.BangBangController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj.motorcontrol.MotorController;
+import edu.wpi.first.wpilibj.motorcontrol.PWMSparkMax;
+import edu.wpi.first.wpilibj.simulation.EncoderSim;
+import edu.wpi.first.wpilibj.simulation.FlywheelSim;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ * This is a sample program to demonstrate the use of a BangBangController with a flywheel to
+ * control RPM.
+ */
+public class Robot extends TimedRobot {
+  private static final int kMotorPort = 0;
+  private static final int kEncoderAChannel = 0;
+  private static final int kEncoderBChannel = 1;
+
+  private final MotorController m_flywheelMotor = new PWMSparkMax(kMotorPort);
+  private final Encoder m_encoder = new Encoder(kEncoderAChannel, kEncoderBChannel);
+
+  // Create Bang Bang controler
+  private final BangBangController m_bangBangControler = new BangBangController();
+
+  // Gains are for example purposes only - must be determined for your own robot!
+  public static final double kFlywheelKs = 0.0001;
+  public static final double kFlywheelKv = 0.000195;
+  public static final double kFlywheelKa = 0.0003;
+  private final SimpleMotorFeedforward m_feedforward =
+      new SimpleMotorFeedforward(kFlywheelKs, kFlywheelKv, kFlywheelKa);
+
+  private final Joystick m_joystick = new Joystick(0); // Joystick to control setpoint
+  private static final double kMaxSetpointValue = 6000; // Max value for joystick control
```

Comment:
What units?

Reply:
So that max setpoint value is slightly above the max free rpm speed of a neo. The units output by the getRate method in the java example is RPM.  (at least for simulation). 
`
m_encoderSim.setRate(m_flywheelSim.getAngularVelocityRPM());
`
I did this because I forgot the behavior of .getRate() by default is different than the getVelocity() Method on the Spark maxes.
I am assuming I should update this java example to use Rad/s as its units. And use the Flywheel sim method of 
getAngularVelocityRadPerSec(). That way both of them match.